### DISCUSSION TITLE
Enhancement: Abiliy to send username to model instead of user id

### DISCUSSION THREAD

**Discussion by alihacks** [danny-avila/LibreChat#4242] [Feature Requests & Suggestions]
### What features would you like to see added?

Currently, calls to the model include user: user.id with the id from the mongo database (for ex. `"user": "66f1aa7e17c9107b86037878"`) . It would be more helpful to send the actual username to the endpoint for reporting across systems (ex. `"user": "alihacks"`) I would be happy to submit a PR for this.

### More details

N/A

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
sure should be optional between the 2 from backward compatibility, defaulting to ID

**Reply by alihacks** [reply]
Sounds good, control this with an environment variable? 
```
SEND_USERNAME_TO_MODEL=true #set to true to send username to model calls, default=false, user id is sent`
```

**Reply by timmanik** [reply]
Hey @alihacks, I'm also interested in this. Were you able to implement it in your fork?

**Reply by alihacks** [reply]
Yes, you can see it in the PR https://github.com/danny-avila/LibreChat/pull/4249

**Reply by timmanik** [reply]
**Edit: I got it to work. Had to edit how Librechat passed the headers to custom endpoints. You can disregard my comment below.**

Thanks @alihacks. I tried it out and am having some trouble getting the username data to my endpoint.

I'm trying to pass the username to my custom endpoint, but it never gets passed through.

I tried {userId}, {user_id}, {user}, and {username}, and none of them gets passed through. Am I using the wrong variable?

Also, what endpoint are you using to pass the usernames to?


Below is my custom endpoint config [1][2] in `librechat.yaml` if you're curious.
```yaml
  custom:
    - name: "Portkey"
      apiKey: "dummy"
      baseURL: "${PORTKEY_GATEWAY_URL}"
      headers:
        x-portkey-api-key: "${PORTKEY_API_KEY}"
        x-portkey-virtual-key: "${PORTKEY_VIRTUAL_KEY}"
        x-portkey-metadata: '{"_user": ${userId}}'
 ```
 
 Reference:
 [1] https://portkey.ai/docs/integrations/libraries/librechat
 [2] https://portkey.ai/docs/api-reference/inference-api/headers#2-metadata-x-portkey-metadata

**Reply by sarangrao** [reply]
@timmanik trying the same and not having much success. Can you add the working config for your endpoint?
I'm using my custom wrapper api. 
I've tried the below
headers:
        X-LibreChat-User: "{{LIBRECHAT_USER_ID}}"

**Reply by rwmehta** [reply]
Somewhat related to this, does anyone know of a way to pass the JWT token generated through google sign in on to the custom endpoint? I'm trying to authenticate to another service my custom endpoint is using with the same sign in.

**Reply by timmanik** [reply]
@sarangrao sorry for the late reply. I missed this message for some reason.

Here's some context. [This is the code](https://github.com/danny-avila/LibreChat/blob/main/api/server/services/Endpoints/custom/initialize.js) that handles the placeholder variables for headers for custom endpoints.

Currently, there is only one type of placeholder variable that can be replaced/used in the `librechat.yaml` file. That placeholder value uses the format of **${variable_name}**. And it pulls the variable from the `.env` file.

In other words, the variables in `librechat.yaml` cannot be dynamically replaced at the moment.

However, see PR #4488. I made it possible to dynamically replace the variable **${userId}** in any custom config header. I linked my endpoint config there too.

Do you have the code for your custom wrapper api?
